Ny komponent FileUpload

torkjels opened the pull request:
Ny komponent FileUpload 🎉 🎉 🎉

torkjels committed:
Ny komponent FileUpload

torkjels committed:
Legg til knapp i box-varianten

JulianNymark commented on @navikt/core/react/src/form/file-upload/FileUpload.tsx:
```diff
@@ -0,0 +1,161 @@
+    if (variant === "button") {
+      return (
+        <div className={className}>
+          <Button
```
Var det endringer for className = 'navds-button...' + Button -> Label ? (eller de var kanskje pending changes 🤔 / spontant i møtet )

torkjels replied:
@JulianNymark  Jeg pusha de endringene nå 😊 Samt en commit hvor jeg skiller på accepted og rejected filer a la slik Polaris gjør det i sin Dropzone.

https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/DropZone/DropZone.tsx

https://polaris.shopify.com/components/selection-and-input/drop-zone

Bygges går rødt, det kan nok ryddes litt i koden, og jeg tror det kan være verdt å skrive noen tester på blant annet dette med accept/reject, men ville bare få pusha det jeg hadde! 😊

torkjels replied:
Tester for accept/reject her https://github.com/navikt/aksel/pull/2504/commits/a794caf6e628d310f920d18ee791c1cc6f91be0b

torkjels committed:
Skill på acceptede og rejectede filer i onUpload

Og leggg på dragover-effekt

torkjels committed:
Ha én return og flere if's for å gjenbruke kode

Usikker på om dette er foretrukket over slik det var, men verdt et forsøk for diskusjonens skyld.

torkjels commented on @navikt/core/react/src/form/file-upload/FileUpload.tsx:
```diff
@@ -124,86 +123,42 @@ export const FileUpload = forwardRef<HTMLDivElement, FileUploadProps>(
+            "navds-fileupload--dragover": isDraggingOver
+          })
+        }>
+          {isBoxVariant && (<>
```
Jeg la til en commit https://github.com/navikt/aksel/pull/2504/commits/4376843ded7f5782a855e321bcba880f98fb355a hvor jeg endret fra å ha én if og tilhørende return per variant og heller kjøre én felles return og gjøre flere if'er inni den returnen for å rendre basert på variant.

Fordelen er at man kan gjenbruke litt mer kode mellom variantene, men ulempen er at det kan være vanskeligere å lese når koden til variantene står om hverandre.

Hva slags pattern foretrekker dere i Aksel?

KenAJoh replied:
Kommer litt an på kompleksiteten på hva if-testen returnerer. Hvis jeg skjønte variant-`box` riktig så er det dropzone området? Om så kan det bli ganske komplekst etterhvert med events hvis vi ønsker forskjellige visninger basert på hover/drag-over state, så kan kanskje trekkes ut til en egen fil 🤔

torkjels replied:
`box` er dropzone-ja. Den eneste forskjellen er at i `box` vises det to tekster til ("Dra og slipp", "eller") og at man legger på en klasse ekstra som gjør at labelet får padding, border og backgroud-color som styler dropzone-området.

Per nå er det ikke så veldig mye logikk basert på variant, så det er ganske deilig å slippe å duplisere kode eller dra alt ut i ekstra sub-komponenter imo.

Vi kan kanskje ta opp diskusjonen igjen når all funksjonaliteten er ferdig og vi vet hva kompleksiteten blir? 🤔 

Jeg har forresten laget en TODO-liste i PR-beskrivelsen, har dere noen innspill der? Dere kan godt bare redigere beskrivelsen direkte hvis dere har rettigheter til det 🙌

torkjels committed:
Dra ut partition-utils i egne filer og skriv tester for dem

torkjels committed:
Fiks brukket bygg pga typescript-feil

torkjels committed:
Fiks brukket bygg ved å fjerne ubrukte props i stories

torkjels committed:
Gjør om label-innhold til span og bruk gap istedenfor margin

Etter prat med Vegard fant vi ut at det var hakket mer semantisk korrekt (og gap er litt penere enn egen klasse med margin)

torkjels committed:
Focus-, hover- og error-styling

torkjels committed:
ondragover-effekt

torkjels committed:
Fiks fokus-markering som ikke funker når viser error

torkjels committed:
Fiks drag-effekt og dra ut i egne sub-komponenter

torkjels committed:
Legg på label og description

Måtte da flytte labelet litt lenger inn og heller legge stylinga på et element lenger ut, men det løste seg.

torkjels committed:
Sett min-width tilsvarende height*2.

Så lenge parent ikke har en mindre width enn det som her blir minWidth vil vi ikke få overflow. Er dette for en for yolo løsning?

KenAJoh commented on @navikt/core/react/src/form/file-upload/Wrapper.tsx:
```diff
@@ -0,0 +1,105 @@
+      style={{
+        width: style?.width ? style.width : fullWidth ? "100%" : "auto",
+        height: style?.height,
+        minWidth: style?.minWidth
```
Utforsket litt alternative løsninger for utfordringen med å beholde høyde ved drag-over. Enkleste løsning jeg fant er å vise et `:after`-element over dropzone slik som f.eks [Mesh har gjort her](https://www.meshdesignsystem.com/components/file-upload-component). Slipper da å blande inn ekstra js-logikk for å sette høyde og kan løses med noen linjer CSS 🤞

KenAJoh commented on @navikt/core/react/src/form/file-upload/FileUpload.tsx:
```diff
@@ -0,0 +1,151 @@
+      event.target.value = ""
+    }
+
+    if (variant === "button") {
```
For å unngå prop-drilling kan et nyttig pattern være
```tsx
<FileUpload>
  // <FileUpload.DropZone />
  // <FileUpload.UploadButton />
</FileUpload>
```

`FileUpload` vil da inneholde state og deler av logikken + være en context-provider
```tsx
// FileUpload.tsx
...logikk
<FileUploadContext.Provider value={...}>
   {children}
</FileUploadContext.Provider>
```

`FileUpload.DropZone` og `FileUpload.UploadButton` blir da en egne komponenter som har egne props og konsumerer context fra `FileUpload`.

torkjels committed:
Fiks lint-errors

torkjels committed:
Rød error-border på zone-variant

torkjels committed:
ondrag: bytt js-løsning med after-løsning

torkjels committed:
Endre fra Wrapper og variant til sub-komponenter

torkjels committed:
Fjern kommentar med feil props-lenke. Variantene har ikke egne props

torkjels committed:
Fiks auto-fyll av bredde

torkjels committed:
Støtte for nb, nn og en locale

torkjels committed:
Småtweak på eksempel og feilmelding

torkjels committed:
Bruk useFormFields

torkjels committed:
Endre knappe-tekst

torkjels committed:
Endre span-knapp til Button-knapp

KenAJoh committed:
Merge branch 'main' into file-uploader

KenAJoh committed:
:truck: Flyttet component-definition over kode

KenAJoh committed:
:truck: Rename context, update exports, moved parts -> /parts

KenAJoh committed:
:bug: Fixed imports in parts

KenAJoh committed:
:label: rename buttonvariant -> uploadbutton

KenAJoh committed:
:recycle: refactor types/forwardref in UploadButton

KenAJoh committed:
:label: Rename zone -> dropzone, :recycle: Refactor dropzone with forwardref

KenAJoh committed:
:label: zone -> dropzone in stories

KenAJoh commented on @navikt/core/react/src/form/file-upload/FileUpload.tsx:
```diff
@@ -189,19 +192,11 @@ export const FileUpload = forwardRef<HTMLDivElement, FileUploadProps>(
   }
 ) as FileUploadComponent;
 
-export interface FileUploadComponent
```
Flyttet denne til toppen av filen, under props for å matche lignende kode i systemet

KenAJoh replied:
Fjernet også export da den ikke trengs å eksponeres

KenAJoh commented on @navikt/core/react/src/form/file-upload/FileUpload.tsx:
De fleste endringene komme fra code-format med prettier 💄

KenAJoh commented on @navikt/core/react/src/form/file-upload/ButtonVariant.tsx:
Så ikke en grunn til å beholde denne, så bruker bare `UploadButton` direkte

KenAJoh commented on .storybook/main.ts:
Fikk ikke kjørt storybook lokalt da denne feiler nå. Kommenterte den bare ut for nå

KenAJoh commented on @navikt/core/react/src/form/file-upload/FileUpload.tsx:
```diff
@@ -0,0 +1,201 @@
+  extends React.ForwardRefExoticComponent<
+    FileUploadProps & React.RefAttributes<HTMLDivElement>
+  > {
+  Dropzone: typeof Dropzone;
```
Renamet zone -> dropzone

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/Dropzone.tsx:
La til props + forwardRef-støtte på Dropzone

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/UploadButton.tsx:
La til props + forwardRef-støtte. La også til muligheten til å legge til egen tekst med `children` da jeg kan se for meg man endrer denne basert på context. F.eks "last opp PDF-er", "Last opp vedlegg" etc

KenAJoh commented on @navikt/core/react/src/form/file-upload/FileUpload.tsx:
```diff
@@ -0,0 +1,201 @@
+import { BodyShort, ErrorMessage, Label } from "../../typography";
+import { useFormField } from "../useFormField";
+import { FileUploadContext } from "./context";
+import Dropzone from "./parts/Dropzone";
```
Flyttet "sub-komponenter" i en egen mappe `parts` for å lette se hva som er "hovedkomponent"

KenAJoh commented on @navikt/core/react/src/form/file-upload/FileUpload.tsx:
```diff
@@ -0,0 +1,201 @@
+          className={cl("navds-form-field", "navds-fileupload", className)}
+          ref={ref}
+        >
+          <Label
```
Kan se for meg at det kunne vært bedre om `<FileUpload/>`-komponenten bare er en contex-provider som tar inn prop-ene 
```
  multiple?: boolean;
  accept?: string;
  onUpload({ allFiles, acceptedFiles, rejectedFiles }: OnUploadProps): void;
  validator?(file: File): boolean;
  locale?: "nb" | "nn" | "en";
```
Disse kan da bli sendt videre gjennom context.
Alt som er innenfor `FileUploadContext.Provider` blir da flyttet til Dropzone som tar inn label, description, error etc

`UploadButton` kan da få sin egen `<input>`, men uten innebygd støtte for label, description og error da dette ikke vil fungere i layout for de flere løsninger. Kan hende label må være med uansett for `<input>`-elementet, og kanskje feilmeldinger også, så må evt utforkses hvordan det håndteres når man har en frittstående knapp for filopplasting. 

Er potensielt en ganske stor endring på strukturen, så tar gjerne en synk på dette

torkjels replied:
Hvorvidt det skal være mulig med standalone knapp uten label/description har kanskje @sjur-gr en formening om? 🤔 
Jeg kan ikke se for meg at man vil bruke knapp (som egentlig betyr knapp og skjult input) uten label og error. Med mindre man vil bruke knappen til å lage en relativt spesiell filopplastings-layout, men isåfall den tid den sorg eller?

KenAJoh replied:
Problemet oppstår hvis man har noe lignende dette layoutet
https://aksel.nav.no/sandbox/index.html?code=N4Igxg9gJgpiBcIA8AhCAPABABwIZSgEsA7AcwF4AdEADmoD5LjNMkAJAZQBdcwBrTKVzYqIAEwMmLFqgCuXLhGYA3XACdCuYl1HYNAW3UBPapkKRi5YEgCq2ADYR8ASQuYAZkq4dCALxiiAIwAdACsajD6pgD09AC%2BjMzSLAAyuADOXJgQ2NgehPYwalLSSNEo8orEiclyCkqYqhpaOtTpMBZQxgwAgsoARmpGXGUV9dUlZZw8-ImjGIkgADQgAO6EUFwAFukIANqBAAxiACwAunFAA

Så vil dette skje med alt rundt knappen
https://aksel.nav.no/sandbox/index.html?code=N4Igxg9gJgpiBcIA8AhCAPABABwIZSgEsA7AcwF4AdEADmoD5LjNMkAJAZQBdcwBrTKVzYqIAEzVMuADaFSxUTGJQGTFiyQA1brwFCR1CSCmz5ogM48ATl1XN1GgDK4ARjGn1nlzBGzYT0jCYAGaEgVZIAPTObh5qDqxoUACeHAAWEDb0ALIwUJhKmLDmYFaE2FyEEMzu4QVWVpmY5hAAtvWYLlZ5MN0FxMx8xMLYSlFJqRlZ8Q6oAK5cXNWYAG64ZbjEXKLYZa3ryZKEkArASACq2NIQ%2BACSJyHVXByEAF4wogCMAHQArN2tSSRegAX0Y9gSmC8XB8fhCYV6M3U4wWS2I4MhSAAog1MrlzOZcKQYPQOABT1ZhIowGGEPh8IIrABnfSuuAJmCsc2UMKGI0KAGpMHNWnNTJgKeYlFxun0%2BX5ejgKbJiAArGBRHGNKz4wnEjHIyLaHj8A0aFCo5ZrDZbCwwE5QA4MACCKy6yS4KMW1QNUU4Jr44PGGHBIAANCAAO6EKBcNLmBAAbU%2BAAYxAAWAC6IKAA

Bruker må derfor evt ha muligheten til å selv plassere label, description og error slik at det kan se slik ut:
https://aksel.nav.no/sandbox/index.html?code=N4Igxg9gJgpiBcIA8AhCAPABABwIZSgEsA7AcwF4AdEADmoD5LjNMkA1AZQBdcwBrTKVzYqIAEwMmLFkgAyuAEYwANvXkBnLpgjZsmXMuUxMAM0JGATkgD08paqnTU0AJ4cAFhAtdGzaSwBZGChMGGZYdTALQmwuQghmFUtQiwsvTHUIAFsUzAULYJgC0OJmPmJhbDDHGWs0KDdPb19-JAAJbl4BIRFqCRB9ZUJSYlEwqEk-Vs4efkFhUX7B4dHqTVxvSf9WlABXLi4EzAA3DcJcYi5RbGisjZdqTEJIUeAkAFVsZQh8AEkX0wJLgcQgALxgogAjAA6ACsBSyj2s9AAvi1tv4NFodHozJYatsbHsDgl0a1rDMumSnMTDsxTtELlc1jAXlB7gwAILHfIuLhE-Z06k2DqzPjCgCiqS8QXU6lwpBg1JYHAApydzJhYFpCHw%2BMZjgAz4pfXByzAWXbEKBacqVMKYADUmF2WV2Q1ImHV6jCXAKxTtuiKOFVBJYQ2IACsYASbFK0hZZfLFeibJT%2BL4iRhfCAADQgADuhBt7nUCAA2pCAAxiAAsAF0UUA

sjur-gr replied:
Kan se for meg at filopplasting kan skje utenfor skjema også. Kommer ikke på noen eksempler fra NAV med det første, men typisk sånn her (last opp profilbilde).

<img width="208" alt="Skjermbilde 2023-11-30 kl  11 20 24" src="https://github.com/navikt/aksel/assets/82579704/84e9bba3-fdad-4ff4-b331-435541ebad54">

KenAJoh replied:
Gir kanskje mening at man kan endre variant av button da også med prop 🤔

torkjels replied:
Er slikt noe man evt kan løse i versjon 2? Det vi har nå vil jeg anta at dekker mer enn brorparten av NAV sine brukstilfeller? Hvis man skal gjøre "alt" på én gang blir man aldri ferdig 😅

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/Dropzone.tsx:
```diff
@@ -0,0 +1,54 @@
+
+    const [minWidth, setMinWidth] = useState<number>();
+
+    useClientLayoutEffect(() => {
```
Tror ikke vi trenger å sette en default høyde/minWidth. Dropzone kan få 100% bredde slik at den tar opp container. 

Kan sette en minheight i CSS på X rem (må teste litt frem hva som fungerer her).

torkjels replied:
Godt poeng. Vet egentlig ikke om vi trenger en minHeight heller, siden høyden bare følger innholdet, og innholdet er basert på tekststørrelsen.

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/Dropzone.tsx:
```diff
@@ -0,0 +1,54 @@
+        <BodyShort as="span">{getDragAndLetGoText(context.locale)}</BodyShort>
+        <BodyShort as="span">{getOrText(context.locale)}</BodyShort>
+        <UploadButton />
+      </div>
```
Er ikke helt sikker på beste løsning for locale-tekster i løsningen, men tror vi trenger noe relativt fleksibelt slik at man kan overskrive dem. Da siden man kanskje vil tilpasse tekster basert på om det en eller flere filer

torkjels replied:
Man kan tilpasse tekster basert på én vs. flere filer basert på `multiple`-propen.

Hvis man likevel vil tilpasse tekster kan man kanskje bare sende inn prop `buttonText` til `FileUpload` og `dragAndDropText: { dragAndDrop: "Dra og slipp", or: "eller" }` til `FileUpload.Dropzone`.

KenAJoh replied:
Kan kanskje teste noe lignende denne syntaksen her 🤔 https://github.com/navikt/aksel/pull/2356/files#:~:text=*/-,translations,-%3F%3A%20ComponentTranslation%5B
Er da bare et object slik som her: https://github.com/navikt/aksel/blob/db02f9451685e1bf2f3ebef54dc562050579ff4b/%40navikt/core/react/src/locales/nb.json

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/UploadButton.tsx:
```diff
@@ -0,0 +1,42 @@
+        ref={ref}
+        className={cl("navds-fileupload__content__button", className)}
+        variant="secondary"
+        onClick={context.onButtonClick}
```
```suggestion
        onClick={(e) => {
            rest.onClick?.(e);
            context.onButtonClick();
        )}
```

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/UploadButton.tsx:
```diff
@@ -0,0 +1,42 @@
+        onClick={context.onButtonClick}
+        tabIndex={-1}
+        icon={
+          <UploadIcon fontSize="1.5rem" focusable={false} aria-hidden={true} />
```
focusable={false} er default i `aksel-icons`
```suggestion
          <UploadIcon fontSize="1.5rem"  aria-hidden />
```

KenAJoh commented on @navikt/core/css/form/file-upload.css:
```diff
@@ -0,0 +1,95 @@
+  position: relative;
+  cursor: pointer;
+  background-color: var(--ac-file-upload-button-bg, var(--a-bg-default));
+  transition: 0.2s ease all;
```
```suggestion
  transition: background-color 0.2s ease;
```

KenAJoh commented on @navikt/core/css/form/file-upload.css:
```diff
@@ -0,0 +1,95 @@
+  cursor: pointer;
+}
+
+.navds-fileupload__content__icon {
```
Ikke brukt lengre

KenAJoh commented on @navikt/core/css/form/file-upload.css:
```diff
@@ -0,0 +1,95 @@
+  transition: 0.2s ease border, 0.2s ease background-color;
+  display: flex;
+  flex-direction: column;
+  gap: 10px;
```
```suggestion
  gap: var(--a-spacing-3);
```

torkjels committed:
Update @navikt/core/css/form/file-upload.css

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

torkjels committed:
Update @navikt/core/css/form/file-upload.css

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

torkjels committed:
Update @navikt/core/react/src/form/file-upload/parts/UploadButton.tsx

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

torkjels committed:
Update @navikt/core/react/src/form/file-upload/parts/UploadButton.tsx

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

torkjels committed:
Fjern ubrukt klasse

torkjels committed:
Fiks parentes-feil

torkjels committed:
Fjern logikk for minWidth ettersom vi uansett har width: 100%

torkjels committed:
Fjern Button-variant for å holde versjon 1 clean

torkjels committed:
Nytt design

torkjels committed:
Juster ned border-width på error for å unngå hopping. Kan fikses senere.

torkjels committed:
Fjern ubrukt styling

torkjels committed:
Merge branch 'main' into file-uploader

# Conflicts:
#	yarn.lock

torkjels committed:
Fiks ref-type

torkjels committed:
Fjern ubrukt context

torkjels committed:
Skjul "Dra og slipp eller" for skjermleser

torkjels committed:
Støtte for å lime inn filer

torkjels committed:
Merge branch 'main' into file-uploader

# Conflicts:
#	yarn.lock

torkjels committed:
Fjern input-element som ikke skulle med

torkjels committed:
Flytt paste-funksjon inn i effect og fjern ekkel for-løkke

torkjels committed:
FileUpload.List (#2535)

torkjels committed:
Fjern endringer i urelaterte filer som kom etter en mislykka merge

torkjels committed:
prettier

torkjels committed:
Legg til word-break

torkjels committed:
La ikon og knapp forbli på toppen ved breaking over flere linjer

torkjels committed:
isNativeFile funker nå også i Node

HalvorHaugan commented on @navikt/core/react/src/form/file-upload/item/ItemName.tsx:
```diff
@@ -0,0 +1,65 @@
+        href={context.href}
+        onClick={(event) => {
+          context.onClick?.(event);
+        }}
```
```suggestion
        onClick={context.onClick}
```

HalvorHaugan commented on @navikt/core/react/src/form/file-upload/list/List.tsx:
```diff
@@ -0,0 +1,62 @@
+          <ul className="navds-file-list__list">
+            {React.Children.map(children, (item) => (
+              <li>{item}</li>
+            ))}
```
Kan sikkert være greit å sjekke om `isValidElement`

torkjels committed:
Native File eller base64 sånn at det funker med ssr

torkjels committed:
PR-pynt

HalvorHaugan commented on @navikt/core/css/tokens.json:
```diff
@@ -357,6 +357,16 @@
+    "--ac-file-upload-error-border": "--a-surface-danger",
+    "--ac-file-upload-action-hover-bg": "--a-surface-action-subtle-hover",
+    "--ac-file-upload-border-radius": "--a-border-radius-large",
+    "--ac-file-upload-icon-bg": "--a-surface-neutral-subtle"
```
Tokens: Revider navngivning. Rydd opp i tokens for button.

HalvorHaugan replied:
@KenAJoh Har du noen tanker om prefix på tokens? De som nå har prefix `file-upload` gjelder jo egentlig dropzone, så kanskje de burde prefixes med `file-upload-dropzone`, eller blir det fryktelig langt? Ikke sikkert vi trenger å ha med "file-upload", siden komponentene egentlig er uavhengige.

(Det er også fire til som ikke synes i kommentaren her, som har prefix `--ac-file-upload-file`)

KenAJoh replied:
`--ac-dropzone` fungerer som prefix det 👍

torkjels committed:
Importér fra navikt/ds-react

torkjels committed:
Revert "Importér fra navikt/ds-react"

This reverts commit 8da51e4c224e6e25fb88fb44dca74fa7d383de61.

torkjels committed:
Endre "Last opp filer" til "Velg filer"

torkjels committed:
Eksportér filtyper

torkjels committed:
Dokumentér base64

torkjels committed:
Bruk riktig border-farge på items

Og bruk tokens

torkjels committed:
Eksportér FileItem

torkjels committed:
Rename onUpload til onSelect

torkjels committed:
Fjern rusk fra stories

torkjels committed:
Margin over og under lista

torkjels committed:
Fiks label på story

HalvorHaugan committed:
Småpirk

torkjels committed:
Oppdatert accept-dokumentasjon

HalvorHaugan committed:
refak item

HalvorHaugan committed:
i18n item

HalvorHaugan committed:
loader margin top

HalvorHaugan committed:
refak file type check

HalvorHaugan committed:
refak formatFileSize

HalvorHaugan committed:
tekst for multiple=false

HalvorHaugan committed:
mappings deps

HalvorHaugan committed:
labelTag og valgfri label

HalvorHaugan committed:
List error css

HalvorHaugan committed:
Justeringer Item og litt List

HalvorHaugan committed:
ID genereres automatisk

HalvorHaugan committed:
isDownloading

HalvorHaugan committed:
Stories!

HalvorHaugan committed:
Slå sammen isUploading og isDownloading

HalvorHaugan committed:
storyflikking

HalvorHaugan committed:
Merge branch 'main' into file-uploader

HalvorHaugan committed:
Endringer

HalvorHaugan committed:
Fokus

HalvorHaugan committed:
extends FormFieldProps

HalvorHaugan committed:
eksport og memo

HalvorHaugan committed:
Bort med Accept

torkjels committed:
Ikke vis error-styling på upload/download

torkjels committed:
Oppdater stories

HalvorHaugan committed:
ikke fjern error border på hover

torkjels committed:
Fjern unødvendig base64-til-blob-konvertering

torkjels committed:
Endre base64 til base64DataUrl

HalvorHaugan committed:
color på btn ved hover, samt rydda litt i tokens

torkjels committed:
Fjern base64DataUrl

KenAJoh committed:
Merge branch 'main' into file-uploader

KenAJoh committed:
:bug: Fikset imports av utils

KenAJoh committed:
:bug: Fikset imports

KenAJoh committed:
:fire: Fjernet ekstra story

KenAJoh committed:
:truck: Flyttet locale-provider til wrapper

KenAJoh committed:
:reacycle: Fjernet children-mappin i List

KenAJoh committed:
:recycle: Oppdatert bruk av locale i komponenter

KenAJoh committed:
:fire: Fjernet nn-locale

KenAJoh committed:
:fire: Fjernet story-snapshot

KenAJoh committed:
:memo: Todo

KenAJoh committed:
:fire: Fjernet listitem

KenAJoh committed:
:fire: Fjernet list

KenAJoh committed:
:fire: Fjernet duplikat error-marking

KenAJoh committed:
:art: Wrapper er nå et div-element

KenAJoh committed:
:art: ul,li-reset

KenAJoh committed:
:lipstick: Justert visning av dragover-overlay

KenAJoh committed:
:lipstick: Drag-n-drop animert

KenAJoh committed:
:lipstick: Animate icon

KenAJoh committed:
:art: Animated demo

KenAJoh committed:
:lipstick: animated box-shadow on error item

KenAJoh committed:
:lipstick: action-selected farge for dragover tekst

KenAJoh committed:
:bug: Fikset locale-bug

KenAJoh committed:
WIP: checkpoint animation

KenAJoh committed:
:lipstick: Animated colors

KenAJoh committed:
:test_tube: Widget story

KenAJoh committed:
Merge branch 'main' into file-uploader

KenAJoh committed:
:fire: Fjernet console.log

KenAJoh committed:
:lipstick: Transition på dragover-text

KenAJoh committed:
:recycle: Forenklet bruk av :has

KenAJoh committed:
:fire: Fjernet bruk av navds-file-dropzone__dragover som selector

KenAJoh committed:
:recycle: Dragover-overlay er nå psudo-element

KenAJoh committed:
:memo: Todo

KenAJoh committed:
:recycle: Oppdatert locale-bruk i fileupload

KenAJoh committed:
:bug: hover på zone oppdaterte ikke button

KenAJoh committed:
:truck: flyttet dropzone til egen mappe

KenAJoh committed:
:truck: Flyttet logikk for dropzone ut i egen hook

KenAJoh committed:
:truck: Trukket ut logikk for dropzone i hook

KenAJoh committed:
:fire: Fjernet FileUpload.List

KenAJoh committed:
:lipstick: CSS refactor

KenAJoh committed:
:fire: Fjernet ref-sjekk for onDrag

KenAJoh committed:
:bug: Pointer-events none for å beholde drag-state

KenAJoh committed:
:recycle: Bruker tokens internt i fileupload css

KenAJoh committed:
:recycle: CSS-naming

KenAJoh committed:
:memo: JSDOC for dropzone and fileupload

KenAJoh committed:
:recycle: FileUpload item støtter overridabecomponent

KenAJoh committed:
:recycle: Forenklet FileUpload.Item

KenAJoh committed:
:recycle: Item sizing

KenAJoh committed:
:lipstick: Small loading-text

KenAJoh committed:
:lipstick: Animated bg on item-icon

KenAJoh committed:
:recycle: Oppdatert conditional-props for FileUpload.Item

KenAJoh committed:
:art: FileError icon on Item-error

KenAJoh committed:
:lipstick: Animert ikonstørrelse

KenAJoh committed:
:lipstick: Padding på FileItem

KenAJoh committed:
:label: rename zone className

KenAJoh committed:
:label: Rename zone -> area

KenAJoh committed:
:memo: FileItem jsdoc

KenAJoh committed:
:truck: Oppdatert types-name

KenAJoh committed:
:memo: Oppdatert locale-tekster

KenAJoh committed:
:bug: Locale-story oppdatert

KenAJoh committed:
:art: Tydeligere border on hover

KenAJoh committed:
:tada: FileUpload.Trigger komponent

KenAJoh committed:
:fire: Fjernet extended inputprops fra Trigger

KenAJoh committed:
:art: Oppdatert feilvalidering

KenAJoh committed:
:bug: Fikset stories

KenAJoh committed:
:bug: Import feilet

KenAJoh committed:
WIP: Disabled-state

KenAJoh committed:
:recycle: Refactor useDropzone -> useFileUpload

KenAJoh committed:
:fire: Fjernet copy-paste feature

KenAJoh committed:
Merge branch 'main' into file-uploader

KenAJoh committed:
:twisted_rightwards_arrows: merge main yarn.lock

KenAJoh committed:
:recycle: Error i Item er nå utenfor box

KenAJoh committed:
:art: Statustekst vises nå sammen med error

KenAJoh committed:
:label: Types

KenAJoh committed:
:art: Custom tekster

KenAJoh committed:
WIP: custom-tekster

KenAJoh committed:
:memo: Bedre storybook-eksempler

KenAJoh committed:
:memo: UU-tester

KenAJoh committed:
:art: La til download attr på item-name

KenAJoh committed:
:recycle: Refactor hvordan status og delete/reset håndteres

KenAJoh committed:
:recycle: Oppdatert typer i story

KenAJoh committed:
Merge branch 'main' into file-uploader

KenAJoh committed:
Merge branch 'main' into file-uploader

KenAJoh committed:
:truck: Kopiert over kode for i18n

KenAJoh committed:
:globe_with_meridians: Locale støtte

KenAJoh committed:
:fire: Første release uten tokens

KenAJoh committed:
:fire: Fjernet redundant stories

KenAJoh committed:
:memo: changeset

KenAJoh committed:
:fire: Fjernet locale JSDOC

HalvorHaugan committed:
Småpirk

HalvorHaugan committed:
Test av reason, test av maxSizeInBytes, rename testfil

KenAJoh commented on @navikt/core/css/form/file-upload.css:
```diff
@@ -0,0 +1,246 @@
+  box-shadow: inset 0 2px 7px 3px rgb(11 11 11/ 0.1);
+  border-radius: calc(var(--a-border-radius-large) - 1px);
+  animation: akselDropzoneDragoverAnimation var(--__ac-dropzone-animation-length-short) var(--__ac-dropzone-animation-ease-out);
+  cursor: grabbing !important;
```
```suggestion
```

KenAJoh committed:
Update @navikt/core/css/form/file-upload.css

KenAJoh committed:
:lipstick: Endret fra border -> outline på dropzone

KenAJoh committed:
:bug: border-color -> outline-color

KenAJoh committed:
:fire: Fjernet place-center på disabled dropzone

KenAJoh committed:
:lipstick: file-item offset is -1px

KenAJoh commented on @navikt/core/react/src/form/file-upload/FileUpload.tsx:
```diff
@@ -0,0 +1,125 @@
+interface FileUploadProps extends HTMLAttributes<HTMLDivElement> {
+  children: React.ReactNode;
+  /**
+   * i18n-API for easier access to customizing texts and labels
```
```suggestion
   * i18n-API for customizing texts and labels
```

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/dropzone/dropzone.types.ts:
```diff
@@ -0,0 +1,21 @@
+   */
+  icon?: React.ComponentType<any>;
+  /**
+   * i18n-API for easier access to customizing texts and labels
```
```suggestion
   * i18n-API for customizing texts and labels
```

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/item/Item.tsx:
```diff
@@ -0,0 +1,168 @@
+   */
+  /* itemAction?: "delete" | "retry" | "none"; */
+  /**
+   * i18n-API for easier access to customizing texts and labels
```
```suggestion
   * i18n-API for customizing texts and labels
```

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/item/Item.tsx:
```diff
@@ -0,0 +1,168 @@
+  /**
+   *
+   */
+  /* itemAction?: "delete" | "retry" | "none"; */
```
```suggestion
```

KenAJoh committed:
Apply suggestions from code review

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/Trigger.tsx:
```diff
@@ -0,0 +1,60 @@
+          onClick={() => {
+            if (inputRef.current?.value) {
+              inputRef.current.value = "";
+            }
```
```suggestion
```

KenAJoh committed:
Update @navikt/core/react/src/form/file-upload/parts/Trigger.tsx

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/dropzone/useDropzone.ts:
```diff
@@ -0,0 +1,35 @@
+  const onDragLeave = () => {
+    if (!isDraggingRef.current) {
+      return;
+    }
```
```suggestion
```

KenAJoh committed:
Update @navikt/core/react/src/form/file-upload/parts/dropzone/useDropzone.ts

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/dropzone/useDropzone.ts:
```diff
@@ -0,0 +1,32 @@
+  const [isDraggingOver, setIsDraggingOver] = useState(false);
+  const isDraggingRef = useRef(false);
+
+  // onDragOver triggers 60+ times per second, so we need to cut it off, and
```
```suggestion
  // onDragOver triggers 60+ times per second, so we cut it off to avoid excessive computation.
```

KenAJoh commented on @navikt/core/react/src/form/file-upload/parts/dropzone/useDropzone.ts:
```diff
@@ -0,0 +1,32 @@
+    setIsDraggingOver(true);
+  };
+
+  // onDragOver triggers 60+ times per second, so we need to cut it off, and
```
```suggestion
```

KenAJoh committed:
Update @navikt/core/react/src/form/file-upload/parts/dropzone/useDropzone.ts

KenAJoh committed:
Update @navikt/core/react/src/form/file-upload/parts/dropzone/useDropzone.ts

HalvorHaugan committed:
JSDOC, css og litt flikking på item stories

HalvorHaugan committed:
Merge branch 'main' into file-uploader

HalvorHaugan committed:
JSDOC, fiks validering, +++

HalvorHaugan committed:
fix test

HalvorHaugan committed:
Sterkere typer for i18n utils

HalvorHaugan committed:
Stories og litt typer

HalvorHaugan committed:
fix story, endre sjekk av disabled i useFileUpload

HalvorHaugan committed:
Merge branch 'main' into file-uploader

HalvorHaugan committed:
Ny struktur på returnerte filer

HalvorHaugan committed:
Fjern "Aksel" namespace på oversettelser

HalvorHaugan committed:
Mindre margin på error

HalvorHaugan committed:
nb.Aksel -> nb

HalvorHaugan committed:
Fiks test validate-files

HalvorHaugan committed:
Fiks i18n

HalvorHaugan committed:
UNSAFE

HalvorHaugan committed:
Forenklet OBJECT_NOTATION_MATCHER

HalvorHaugan committed:
Eksempler

KenAJoh committed:
:lipstick: animert og oppdatert fileupload error

HalvorHaugan committed:
Merge branch 'main' into file-uploader

HalvorHaugan committed:
exports

HalvorHaugan committed:
fixes

HalvorHaugan committed:
tester noe

HalvorHaugan committed:
fiks playroom

HalvorHaugan committed:
export FileAccepted og RejectedPartitionedFile

HalvorHaugan committed:
rename og export PartitionedFiles

HalvorHaugan committed:
rename RejectedPartitionedFile

HalvorHaugan committed:
ul/li

HalvorHaugan committed:
Trigger button secondary

HalvorHaugan committed:
Justert tekster på default eksempel

HalvorHaugan committed:
Eksempel på Trigger

HalvorHaugan committed:
fix mobile view på story

HalvorHaugan committed:
Ny design på Item error

HalvorHaugan committed:
story på drag state

HalvorHaugan committed:
Merge branch 'main' into file-uploader

HalvorHaugan committed:
fjern rest på Trigger

HalvorHaugan committed:
FileUpload: display:none på input (#2773)

HalvorHaugan committed:
Merge branch 'main' into file-uploader

HalvorHaugan commented on @navikt/core/react/src/form/file-upload/parts/dropzone/dropzone.types.ts:
```diff
@@ -0,0 +1,22 @@
+  /**
+   * i18n-API for customizing texts and labels
+   */
+  translations?: ComponentTranslation["FileUpload"];
```
Burde ikke `translations` prop på Dropzone være `ComponentTranslation["FileUpload"]["dropzone"]`?

HalvorHaugan replied:
Tilsvarende på Item.

KenAJoh replied:
`export type ComponentTranslation = RecursivePartial<typeof nb>;` støtter ikke ComponentTranslation["FileUpload"]["dropzone"] akkurat nå, men har gitt mening å tilby en type som løser dette 👌

HalvorHaugan replied:
Må bli sånn da?
`export type FileUploadTranslation = RecursivePartial<typeof nb.FileUpload>;`
Gir det mest mening å legge denne i i18n.types.ts eller FileUpload.types.ts?

HalvorHaugan committed:
Endringer på i18n

HalvorHaugan committed:
Merge branch 'main' into file-uploader

HalvorHaugan committed:
fix replacement type

HalvorHaugan committed:
i18n tests

HalvorHaugan committed:
Record

KenAJoh reviewed (approved):
De siste endringene så bra ut de! Da tror jeg ikke det er så mye mer som gir mening å teste hos oss før vi slipper den og lar alle andre teste den da 🚀 🚀 💯 

Note for fremtiden.:  Istedenfor å måtte ha contexter som dette: https://github.com/navikt/aksel/pull/2504/files#diff-ac4a6d8d97aaab6c44760b05e5e32a49f446a6c5952470cae310e2c357090cebR129
vil det kanskje gi mening å ha en egen TranslationProvider vi kan bruke internt i systemet.
